worker/uniter/runner/runner.go

```diff
@@ -61,29 +63,71 @@ type Context interface {
 }
 
 // NewRunner returns a Runner backed by the supplied context and paths.
-func NewRunner(context Context, paths context.Paths) Runner {
-	return &runner{context, paths}
+func NewRunner(context Context, paths context.Paths, remoteExecutor ExecFunc) Runner {
+	return &runner{context, paths, remoteExecutor}
 }
 
+// ExecParams holds all the necessary parameters for ExecFunc.
+type ExecParams struct {
+	Commands      []string
+	Env           []string
+	WorkingDir    string
+	Clock         clock.Clock
+	ProcessSetter func(context.HookProcess)
+	Cancel        <-chan struct{}
+	Stdout        io.ReadWriter
+	Stderr        io.ReadWriter
+}
+
+// execOnMachine executes commands on current machine.
+func execOnMachine(params ExecParams) (*utilexec.ExecResponse, error) {
+	command := utilexec.RunParams{
+		Commands:    strings.Join(params.Commands, " "),
+		WorkingDir:  params.WorkingDir,
+		Environment: params.Env,
+		Clock:       params.Clock,
+	}
+	err := command.Run()
+	if err != nil {
+		return nil, err
+	}
+	// TODO: refactor kill process and implement kill for caas exec.
```

Comment:
Can you please add a trello card for this

Reply:
added a card https://trello.com/c/q0voaJtf